contracts/core/Dispatcher.sol

```diff
@@ -384,7 +384,12 @@ contract Dispatcher is IbcDispatcher, IbcEventsEmitter, Ownable, Ibc {
 
         // Not timeout yet, then do normal handling
         IbcPacket memory pkt = packet;
-        AckPacket memory ack = receiver.onRecvPacket(pkt);
+        AckPacket memory ack;
+        try receiver.onRecvPacket(pkt) returns (AckPacket memory receivedAck) {
+            ack = AckPacket(receivedAck.success, receivedAck.data);
+        } catch {
+            ack = AckPacket(false, ""); // Since the call to receiver.onRecvPacket failed; we have an empty ack data
```

Comment:
Maybe we can extract some useful error data from the exception? The `AckPacket.data` will be interpreted as an "error message" in case `success == false`.

Maybe we can implement some of these https://docs.soliditylang.org/en/latest/control-structures.html#try-catch? what do you think?

Reply:
Ah yes this is a good idea! Thanks for calling this out; this is addressed in the new commit !